### PR TITLE
Setup of a few things

### PR DESCRIPTION
**Pull request by hungrysushi**
Added a top level CMakeLists
Added Gtest module
Set up barebones renderer directory

### CODE REVIEW

File: renderer/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (ktadiko) [inline renderer/CMakeLists.txt]: It might be better to put these in a directory called "engine"

**Review by ktadiko — CHANGES_REQUESTED** [review]
Everything looks good. Just put the renderer under engine.

File: renderer/CMakeLists.txt (hunk 1 above)
Comment (ktadiko) [reply renderer/CMakeLists.txt]: Maybe "engine/graphics/renderer" even. Cause shaders, sprites etc could be added there too.

Comment (hungrysushi) [reply renderer/CMakeLists.txt]: I'm fine with putting it under engine, but nesting it 3 levels deep might be too much. Besides, I think sprites and other resources should really be in their own top level folder

**Review by ktadiko — APPROVED** [review]

**Pull request merged**